pinot-transport/src/test/java/com/linkedin/pinot/routing/builder/LargeClusterRoutingTableBuilderTest.java

```diff
@@ -0,0 +1,260 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.routing.builder;
+
+import com.linkedin.pinot.common.utils.CommonConstants;
+import com.linkedin.pinot.routing.ServerToSegmentSetMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.helix.model.ExternalView;
+import org.apache.helix.model.InstanceConfig;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertTrue;
+
+
+/**
+ * Test for the large cluster routing table builder.
+ */
+public class LargeClusterRoutingTableBuilderTest {
```

Comment:
You also need to print the random seed, so that we can set it to the specific value and rerun the tests if it fails :)